cvat/apps/lambda_manager/views.py

```diff
@@ -525,16 +528,30 @@ def enqueue(self,
         *,
         job: Optional[int] = None
     ) -> LambdaJob:
-        jobs = self.get_jobs()
+        queue = self._get_queue()
+        rq_id = RQId(RequestAction.AUTOANNOTATE, RequestTarget.TASK, task).render()
+
         # It is still possible to run several concurrent jobs for the same task.
         # But the race isn't critical. The filtration is just a light-weight
         # protection.
-        if list(filter(lambda job: job.get_task() == task and not job.is_finished, jobs)):
+        rq_job = queue.fetch_job(rq_id)
+
+        have_conflict = rq_job and not rq_job.is_finished and not rq_job.is_failed
```

Comment:
Using `is_finished` and `is_failed` refreshes job status on each action. Since we check that immediately after fetching rq job, we can rely on already obtained status (rq_job.get_status(refresh=False))

Reply:
Thanks, I hadn't realized that. I decided to replace the check entirely with `ended_at is None`, which doesn't re-fetch anything and IMO communicates the intention better.

Reply:
Bleh, turns out `ended_at` isn't set correctly in sync mode, which we use in unit tests: rq/rq#2119. Reimplemented with `get_status(refresh=False)`.